Flipper/Packages/UI/Sources/Common/Image/SVG/SVGParser.swift

```diff
@@ -0,0 +1,181 @@
+        switch elementName {
+        case "path":
+            parsePathElement(attributes: attributeDict)
+        case "rect":
+            parseRectElement(attributes: attributeDict)
+        default: break
+        }
+    }
+
+    private func parsePathElement(attributes: [String: String]) {
+        guard let dAttribute = attributes["d"] else { return }
+
+        let rawCommands = parseRawCommands(data: dAttribute)
+        let svgCommands = rawCommands.map(parseSVGCommand)
+
+        self.commands += svgCommands
+    }
+
+    private func parseRawCommands(data: String) -> [String] {
+        let supportedTags = "MmLlHhVvZzC"
+
+        var result = [String]()
+        var currentCommand = ""
+
+        for character in data {
+            if supportedTags.contains(character) {
+                if !currentCommand.isEmpty {
+                    result.append(currentCommand)
+                }
+                currentCommand = String(character)
+            } else {
+                currentCommand.append(character)
+            }
+        }
+
+        if !currentCommand.isEmpty {
+            result.append(currentCommand)
+        }
+
+        return result
+    }
+
+    private func parseSVGCommand(command: String) -> SVGCommand {
+        guard let commandCharacter = command.first else { return .unknown }
+
+        let value = command
+            .dropFirst()
+            .split(separator: " ")
+            .compactMap{ Double($0) }
+
+        switch commandCharacter {
+        case "M":
+            let point = CGPoint(x: value[0], y: value[1])
+            return .moveAbsolute(point: point)
+        case "m":
+            return .moveRelative(x: value[0], y: value[1])
+        case "L":
+            let point = CGPoint(x: value[0], y: value[1])
+            return .lineToAbsolute(point: point)
+        case "l":
+            return .lineToRelative(x: value[0], y: value[1])
+        case "H":
+            return .horizontalLineToAbsolute(x: value[0])
+        case "h":
+            return .horizontalLineToRelative(x: value[0])
+        case "V":
+            return .verticalLineToAbsolute(y: value[0])
+        case "v":
+            return .verticalLineToRelative(y: value[0])
+        case "C":
+            let x = CGPoint(x: value[0], y: value[1])
+            let y = CGPoint(x: value[2], y: value[3])
+            let to = CGPoint(x: value[4], y: value[5])
+            return .curveToAbsolute(x: x, y: y, to: to)
+        case "Z", "z":
+            return .closePath
+        default:
+            return .unknown
+        }
+    }
+
+    private func parseRectElement(attributes: [String: String]) {
+        guard
+            let width = attributes["width"], let width = Double(width),
+            let height = attributes["height"], let height = Double(height)
+        else { return }
+
+        let x = Double(str: attributes["x"]) ?? 0.0
+        let y = Double(str: attributes["y"]) ?? 0.0
+        if x.isZero && y.isZero { return }
```

Comment:
does it make sense to render if just one of the axes is zero? (_also guard fits better here_)

Reply:
> does it make sense to render if just one of the axes is zero?

yes, if one of them not found, its mean zero value by default
but, if both not found, its mean bad <rect> parse